eth/query/eth.go

```diff
@@ -95,11 +95,28 @@ func getBlockLogs(
 	bloomFilter := evmAuxStore.GetBloomFilter(height)
 	if len(bloomFilter) > 0 {
 		if MatchBloomFilter(ethFilter, bloomFilter) {
-			txHashList, err := evmAuxStore.GetTxHashList(height)
+			var txHashList [][]byte
+
+			// Try get txHashList from BlockStore
+			txObject, err := GetBlockByNumber(blockStore, state, int64(height), false, evmAuxStore)
+			if err == nil {
+				for _, txHash := range txObject.Transactions {
+					hash, err := eth.DecDataToBytes(txHash.(eth.Data))
+					if err != nil {
+						return nil, errors.Wrapf(err, "unable to decode txhash %x", txHash)
+					}
+					txHashList = append(txHashList, hash)
+				}
+			}
 
-			if err != nil {
-				return nil, errors.Wrapf(err, "txhash for block height %d", height)
+			// Try get txHashList from EvmAuxStore instead if txHashList is empty
+			if len(txHashList) == 0 {
+				txHashList, err = evmAuxStore.GetTxHashList(height)
```

Comment:
So the issue here seems to be that the tx hash list has duplicate tx hashes because clusterkit didn't fix the hash lists when it recreated the receipts?

Reply:
Done